api/adaptive_load/metric_spec.proto

```diff
@@ -0,0 +1,41 @@
+// Protos for identifying metrics and specifying thresholds.
+
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "envoy/config/core/v3/extension.proto";
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// Identifies a feedback metric.
+message MetricSpec {
+  // Name of the metric to evaluate. For the set of built-in metric names, see
+  // source/adaptive_load/metrics_plugin_impl.cc. Required.
+  string metric_name = 1 [(validate.rules).string.min_len = 1];
```

Comment:
How do plugins and metric names relate? Does each plugin only export out a fixed number of metric names? Or is metric name an opaque ID?

Reply:
Each plugin will support its own fixed set of metric names. We will query each plugin for its metric names at startup in order to validate the MetricSpec in the adaptive load session input proto.

For nighthawk.builtin:
- latency-min
- latency-mean
- latency-mean-plus-1stdev
- latency-mean-plus-2stdev
- latency-mean-plus-3stdev
- latency-max
- achieved-rps
- attempted-rps
- send-rate
- success-rate

Reply:
SGTM. I'm going to be OOTO next two weeks, so please go ahead with this set of protos. I think as long as we're the only consumers, which is highly likely for the near future, it's fine to iterate as needed.